### ISSUE TITLE
move osinfo logic into tracee (or expand it in libbpfgo ?)

### ISSUE THREAD

**Issue by rafaeldtinoco** [aquasecurity/libbpfgo#297]
I think OSinfo logic is only used by tracee and, it is not part of libbpf nor interesting to be kept inside libbpfgo (IMO). What do you think @geyslan ? We could create a pkg there and get rid of the helper:

https://github.com/aquasecurity/libbpfgo/blob/main/helpers/osinfo.go

from here. It would be easier to expand that pkg there, while removing the code footprint here.

WDYT ?

**Comment by yanivagman** [comment]
This can be a good idea, but we better verify that the users of libbpfgo really don't use it. 
We can open a discussion about that to see if there are any objections

**Comment by yanivagman** [comment]
Actually, we can copy the packages into tracee (including helpers?) and write a comment that it is deprecated in libbpfgo, and whoever wants can use the package from Tracee, WDYT?

**Comment by rafaeldtinoco** [comment]
Yep, good idea, lets deprecate it in libbpfgo for one or two releases and move it into tracee!

**Comment by geyslan** [comment]
> Yep, good idea, lets deprecate it in libbpfgo for one or two releases and move it into tracee!

+1

> Actually, we can copy the packages into tracee (including helpers?) and write a comment that it is deprecated in libbpfgo, and whoever wants it can use the package from Tracee, WDYT?

I think we have a current difficulty in maintaining different packages inside tracee, for example it's laborious to make changes in `types`. We just avoided turning pkg/logger into a pkg of its own by thinking about that trouble, what resulted in a three layer callback.

I guess @NDStrahilevitz has thoughts concerning that as well.

**Comment by rafaeldtinoco** [comment]
> I think we have a current difficulty in maintaining different packages inside tracee, for example it's laborious to make changes in types.

I thought about the multi-packaged tracee (pkg/xxx) package.

**Comment by geyslan** [comment]
https://github.com/aquasecurity/tracee/pull/4090
